rfcs/0004-releases.md

```diff
@@ -57,9 +57,12 @@ The following steps would need to be taken to implement this from a technical pe
   * Add link to update guide in release tag on GH
   * Release triggers a slack webhook to notify that a new release is ready with link to update guide
 
+## Schedule
+* Tradeoffs exist around releases. Too long of a period between releases can potentially cause friction for teams who need new features quickly and are forced to termporarily use the next branch. On the other hand, too short and we increase "update noise" for teams. Additionally, less frequent releases introduce more risk when updating the Eik aliases.
+* A weekly schedule allows for a quick enough turn around such that most teams would not need to use the next branch (up to a week wait at most) while preventing ongoing development noise and allowing for clean communication of new releases. Eg. "New releases of Fabric every Monday at 10am"
```

Comment:
I'm not really sure I get why a schedule is needed. For bug-fixes and the like, it should just be released into master and released automatically at once.

For major changes that should be released when it is ready and not "rushed to Monday or we lose the window."

Reply:
I think that's fair. If we simply work against the next branch (importantly including renovate PRs there) and then just merge to main whenever necessary (which would mean whenever there's a bug fix or new feature) then I think we'd be 99% there. 

What I haven't made clear in the doc here is that we also need to update the Eik alias for each library as well and that we would like to add a layer of safety into this in the form of a delay (maybe a week). This is because when we update the alias for a library, everyone will be updated to the latest version of the library in production without the need to redeploy (and therefore rerun CI). We can roll this back in the case of issues but currently there's a delay as caches expire and such so its not as quick a rollback as in pipeline. We never bump aliases across major version boundaries so in theory the risk is low, but immediately rolling out alias changes whenever we publish a new version of say Fabric React risks that someone make a mistake and some bad code snuck out before we could catch it. Our initial idea was that a delay of one week from release would give plenty of time to detect any issues and fix before the alias was updated. A fixed release schedule makes this lagging alias update process easier to manage by avoiding the need for book keeping around how long a week is from the last release but that's not necessarily a show stopper.